src/classes/RD2_StatusMappingSettings_CTRL.cls

```diff
@@ -0,0 +1,429 @@
+                    mappings.add(new Mapping(mapping));
+                }
+            }
+
+            //get other mapping records next
+            for (String status : mappingByStatus.keySet()) {
+                if (!RD2_StatusMapper.READ_ONLY_STATUS_VALUES.contains(status)) {
+                    mappings.add(new Mapping(mappingByStatus.get(status)));
+                }
+            }
+
+        } catch (Exception e) {
+            throwAuraHandledException(e.getMessage());
+        }
+
+        return mappings;
+    }
+
+
+    /**
+     * @description Saves status to state mapping records
+     * @param jsonMapping JSON representation of the admin defined mapping records
+     * @return String Deployment Id
+     */
+    @AuraEnabled
+    public static String saveMapping(String jsonMapping) {
+        return saveMapping(jsonMapping, new DeploymentHandler());
+    }
+
+    /**
+     * @description Saves status to state mapping records
+     * @param jsonMapping JSON representation of the admin defined mapping records
+     * @param deploymentHandler Deployment handler used to deploy CMT records and return the deployment Id
+     * @return String Deployment Id
+     */
+    @TestVisible
+    private static String saveMapping(String jsonMapping, MetadataDeploymentHandler deploymentHandler) {
+        try {
+            if (deploymentHandler.isInProgress()) {
+                throw new MappingException(System.Label.RD2_StatusMappingDeploymentAlreadyInProgress);
+            }
+
+            List<Metadata.CustomMetadata> records = buildCustomMetadataRecords(jsonMapping);
+            deploymentHandler.save(records);
+
+            //reset mapping record retrieval
+            statusMapper = null;
+
+        } catch (Exception e) {
+            throwAuraHandledException(e.getMessage());
+        }
+
+        return deploymentHandler.getDeploymentId();
+    }
+
+    /**
+     * @description Builds custom metadata records from status to state mapping records
+     * @param jsonMapping JSON representation of mapping records
+     * @return List<Metadata.CustomMetadata>
+     */
+    @TestVisible
+    private static List<Metadata.CustomMetadata> buildCustomMetadataRecords(String jsonMapping) {
+        List<Mapping> mappings = (List<Mapping>) JSON.deserialize(
+            jsonMapping, List<Mapping>.class
+        );
+
+        List<Metadata.CustomMetadata> records = new List<Metadata.CustomMetadata>();
+        for (Mapping mapping : mappings) {
+            records.add(buildCustomMetadata(mapping));
+        }
+
+        return records;
+    }
+
+    /**
+     * @description Builds custom metadata type record for the status to state mapping
+     * @return Metadata.CustomMetadata
+     */
+    private static Metadata.CustomMetadata buildCustomMetadata(Mapping mapping) {
+        String sObjectName = RecurringDonationStatusMapping__mdt.getSObjectType().getDescribe().getName();
+
+        Metadata.CustomMetadata cmt = new Metadata.CustomMetadata();
```

Comment:
I'm not sure you need to apply the time to the label, plus it looks a bit odd if you're viewing this in Salesforce Setup. The Label doesn't need to be unique anyway. I recommend the following:
- Label: PicklistValueApiName + " -> " + State (ex: `No_Response -> Closed`)

The `Name` field is unique and I like that it has the picklist value api name.

<img width="540" alt="image" src="https://user-images.githubusercontent.com/801017/81840439-ff1aff80-9516-11ea-810e-a9d5a2d74eb9.png">